RInAppMessaging/Classes/CampaignDispatcher.swift

```diff
@@ -75,14 +76,24 @@ internal class CampaignDispatcher: CampaignDispatcherType {
                     Error: Campaign (\(campaign.id)) does not exist in the repository anymore (race condition?). Proceeding with old data...
                     """)
             }
-            self.router.displayCampaign(campaign) { [weak self] in
+
+            let campaignTitle = campaign.data.messagePayload.title
+            self.router.displayCampaign(campaign, confirmation: {
+                let contexts = campaign.contexts
+                guard let delegate = self.delegate, !contexts.isEmpty, !campaign.data.isTest else {
+                    return true
+                }
+
+                return delegate.shouldShowCampaignMessage(title: campaignTitle,
+                                                          contexts: contexts)
+            }, completion: { [weak self] in
                 guard let strongSelf = self else {
                     return
                 }
```

Comment:
You can use this instead:
`guard let `self` = self else { return }`

Reply:
But then I'd have to use ``self`...`?

Reply:
Ok I'll use `guard let self = self else { return }` since it's possible since Swift 4.2 👍